From 650f0f7ecfaee42b3da89a04b92b05f27fe786e9 Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Sat, 10 Apr 2021 12:15:54 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick6/issues/145

---
 magick/annotate.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/magick/annotate.c b/magick/annotate.c
index 29c8bbe74..20fbf7bb1 100644
--- a/magick/annotate.c
+++ b/magick/annotate.c
@@ -1484,6 +1484,15 @@ static MagickBooleanType RenderFreetype(Image *image,const DrawInfo *draw_info,
   metrics->pixels_per_em.y=face->size->metrics.y_ppem;
   metrics->ascent=(double) face->size->metrics.ascender/64.0;
   metrics->descent=(double) face->size->metrics.descender/64.0;
+  if (face->size->metrics.ascender == 0)
+    {
+      /*
+        Sanitize buggy ascender and descender values.
+      */
+      metrics->ascent=face->size->metrics.y_ppem;
+      if (face->size->metrics.descender == 0)
+        metrics->descent=face->size->metrics.y_ppem/-3.5;
+    }
   metrics->width=0;
   metrics->origin.x=0;
   metrics->origin.y=0;
-- 
2.31.0

