From 24a62f1609f0a004428b0342a0810208059d984a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Wed, 14 Dec 2016 15:26:15 +0100
Subject: Fix www/command-line-options.html

Forwarded: Yes, https://github.com/ImageMagick/ImageMagick/issues/304
---
 www/command-line-options.html | 1 +
 1 file changed, 1 insertion(+)

diff --git a/www/command-line-options.html b/www/command-line-options.html
index 88cb653ad0..a6ee90cc29 100644
--- a/www/command-line-options.html
+++ b/www/command-line-options.html
@@ -2245,6 +2245,7 @@ available:</p>
 
   <tr>
     <td>png:exclude-chunk=<var>value</var></td>
+  </tr>
 
   <tr>
     <td>png:include-chunk=<var>value</var></td>
