From f7cdaac529f6ab9d331cc5e7dfccc3797efa2a40 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Wed, 14 Dec 2016 15:26:15 +0100
Subject: Fix www/command-line-options.html

Forwarded: Yes, https://github.com/ImageMagick/ImageMagick/issues/304
---
 www/command-line-options.html | 1 +
 1 file changed, 1 insertion(+)

diff --git a/www/command-line-options.html b/www/command-line-options.html
index 9df2d5243..04795ed5d 100644
--- a/www/command-line-options.html
+++ b/www/command-line-options.html
@@ -2215,6 +2215,7 @@ available:</p>
 
   <tr>
     <td>png:exclude-chunk=<var>value</var></td>
+  </tr>
 
   <tr>
     <td>png:include-chunk=<var>value</var></td>
