From: Cristy <mikayla-grace@urban-warrior.org>
Date: Fri, 3 Sep 2021 19:45:32 -0400
Subject: Fix a non initialized value passed to TIFFGetField()

bug-oss-fuzz: https://oss-fuzz.com/testcase-detail/6502669439598592
bug: https://github.com/ImageMagick/ImageMagick6/issues/246
origin: https://github.com/ImageMagick/ImageMagick6/commit/995de330310dd35531165d9471fe4d31e0fa79ae
---
 coders/tiff.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/tiff.c b/coders/tiff.c
index c326559..fd0169f 100644
--- a/coders/tiff.c
+++ b/coders/tiff.c
@@ -1485,6 +1485,7 @@ static Image *ReadTIFFImage(const ImageInfo *image_info,
       }
     if (TIFFGetFieldDefaulted(tiff,TIFFTAG_ORIENTATION,&orientation,sans) == 1)
       image->orientation=(OrientationType) orientation;
+    chromaticity=(float *) NULL;
     if (TIFFGetField(tiff,TIFFTAG_WHITEPOINT,&chromaticity) == 1)
       {
         if ((chromaticity != (float *) NULL) && (*chromaticity != 0.0))
