From: Cristy <mikayla-grace@urban-warrior.org>
Date: Fri, 17 Dec 2021 14:05:04 -0500
Subject: Fix buffer overrun in TIFF coder

origin: https://github.com/ImageMagick/ImageMagick6/commit/add9cb14e14eef02806715d97abcf5d04a3e55dd
---
 coders/tiff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/tiff.c b/coders/tiff.c
index 6c68f8a..102b2b8 100644
--- a/coders/tiff.c
+++ b/coders/tiff.c
@@ -1242,7 +1242,7 @@ static Image *ReadTIFFImage(const ImageInfo *image_info,
     *pixels;
 
   void
-    *sans[4] = { NULL, NULL, NULL, NULL };
+    *sans[5] = { NULL, NULL, NULL, NULL, NULL };
 
   /*
     Open image.
